vpr/src/route/route_timing.cpp

```diff
@@ -1286,11 +1287,11 @@ static bool timing_driven_route_sink(
      * add this branch to the existing route tree and update pathfinder costs and rr_node_route_inf to reflect this */
     auto& route_ctx = g_vpr_ctx.mutable_routing();
     auto& cluster_ctx = g_vpr_ctx.clustering();
-
+    auto& device_ctx = g_vpr_ctx.device();
```

Comment:
Should we use ``const auto&`` since you are not going to modify anything in device modeling.

Reply:
I think auto will depict it as` const` as `device()`is returning `const`.

Reply:
@baberali-rs Do you any article to support this conclusion? It may be interpreted differently by different compilers and we do not want to risk.